### PR TITLE
Add containers

### PR DESCRIPTION
**Pull request by zancas**
@pjbreaux this is a first draft of the test flow tools.

There's too much code here to wait until I get full TRTL/nightly integration.

### CODE REVIEW

File: f5_os_test/testrunner_containers/docker_manager.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    infname = join(kwargs['test_type'], 'project_docker.tmpl')
+    outdir = join(kwargs['test_type'], kwargs['project'])
+    outfname = join(outdir, 'Dockerfile')
+    logging.debug('outfname: {}'.format(outfname))
+    open(outfname, 'w').write(
+        jinja2.Template(open(infname).read()).render(**kwargs)
+    )
+
+def _publish_testrunner_container(registry_fullname):
+    pubstring = "docker push {}".format(registry_fullname)
```
Comment (pjbreaux) [inline f5_os_test/testrunner_containers/docker_manager.py:46]: Think about using docker-py (or some other Python lib for docker) in the future. Might be cool to checkout this:
https://docker-py.readthedocs.io/en/latest/api/

File: f5_os_test/testrunner_containers/docker_manager.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                user=user,
+                image=image_id)
+    logger.debug('registry_fullname: {}'.format(registry_fullname))
+    tag_command = "docker tag {LOCAL} {FULL}".format(LOCAL=local_tag,
+                                                     FULL=registry_fullname)
+    subprocess.check_call(tag_command.split())
+    _publish_testrunner_container(registry_fullname)
+
+
+def main():
```
Comment (pjbreaux) [inline f5_os_test/testrunner_containers/docker_manager.py:92]: Think about docopt or argparse or whatever new flavor of the week for argument parsing.

File: f5_os_test/testrunner_containers/docker_manager.py (hunk 1 above)
Comment (zancas) [reply f5_os_test/testrunner_containers/docker_manager.py:46]: ACK

Comment (zancas) [reply f5_os_test/testrunner_containers/docker_manager.py:46]: I agree that this is a good idea, I'll write an enhancement issue for it.

File: f5_os_test/testrunner_containers/tempest/project_docker.tmpl [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+RUN pip install pytest
+COPY ./pytest-autolog ./pytest-autolog
+RUN pip install ./pytest-autolog
+COPY ./pytest-symbols ./pytest-symbols
+RUN pip install ./pytest-symbols
+COPY ./pytest-meta ./pytest-meta
+RUN pip install ./pytest-meta
+
+# cloning tempest and F5 neutron-lbaas. Sanaz, why do we point at Mitaka?
+RUN git clone -b stable/mitaka https://github.com/F5Networks/neutron-lbaas.git
```
Comment (pjbreaux) [inline f5_os_test/testrunner_containers/tempest/project_docker.tmpl:50]: Parameterize branch or tag

File: f5_os_test/testrunner_containers/docker_manager.py (hunk 2 above)
Comment (zancas) [reply f5_os_test/testrunner_containers/docker_manager.py:92]: I intend to use argparse in the next iteration.

File: f5_os_test/testrunner_containers/tempest/project_docker.tmpl (hunk 3 above)
Comment (zancas) [reply f5_os_test/testrunner_containers/tempest/project_docker.tmpl:50]: Also on the road map.

File: f5_os_test/testrunner_containers/tempest/set_tempest_config.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+crudini --set $TEMPEST_CONF network public_router_id $PUBLIC_ROUTER_ID
+crudini --set $TEMPEST_CONF network public_network_id $PUBLIC_NETWORK_ID
+crudini --set $TEMPEST_CONF identity uri $OS_AUTH_URL
+crudini --set $TEMPEST_CONF identity uri_v3 $OS_AUTH_URL_V3
+crudini --set $TEMPEST_CONF auth admin_tenant_id $OS_TENANT_ID
+crudini --set $TEMPEST_CONF f5_lbaasv2_driver icontrol_hostname $ICONTROL_IPADDR
+crudini --set $TEMPEST_CONF f5_lbaasv2_driver icontrol_username admin
+crudini --set $TEMPEST_CONF f5_lbaasv2_driver icontrol_password admin
+crudini --set $TEMPEST_CONF f5_lbaasv2_driver transport_url rabbit://guest:guest@$CONTROLLER_IPADDR:5672/
```
Comment (pjbreaux) [inline f5_os_test/testrunner_containers/tempest/set_tempest_config.sh:26]: Parameterize the rabbit username and passwords (nice to have)

**Comment by zancas** [comment]
Not gonna' happen.   This approach has been deprecated.

**Pull request closed without merging**